src/event_trigger.c

```diff
@@ -63,11 +69,123 @@ event_trigger_ddl_commands(void)
 	return objects;
 }
 
+/* Given a TEXT[] of addrnames return a list of char *
+ *
+ * similar to textarray_to_strvaluelist */
+static List *
+extract_addrnames(ArrayType *arr)
+{
+	Datum	   *elems;
+	bool	   *nulls;
+	int			nelems;
+	List	   *list = NIL;
+	int			i;
+
+	deconstruct_array(arr, TEXTOID, -1, false, 'i',
+					  &elems, &nulls, &nelems);
+
+	for (i = 0; i < nelems; i++)
+	{
+		if (nulls[i])
+			elog(ERROR, "unexpected null in name list");
+		list = lappend(list, TextDatumGetCString(elems[i]));
+	}
+
+	return list;
+}
+
+List *
+event_trigger_dropped_objects(void)
+{
+	ReturnSetInfo rsinfo;
+	FunctionCallInfoData fcinfo;
+	TupleTableSlot *slot;
+	EState	   *estate = CreateExecutorState();
+	List	   *objects = NIL;
+
+	InitFunctionCallInfoData(fcinfo, &dropped_objects_fmgrinfo, 0, InvalidOid, NULL, NULL);
+	MemSet(&rsinfo, 0, sizeof(rsinfo));
+	rsinfo.type = T_ReturnSetInfo;
+	rsinfo.allowedModes = SFRM_Materialize;
+	rsinfo.econtext = CreateExprContext(estate);
+	fcinfo.resultinfo = (fmNodePtr) &rsinfo;
+
+	FunctionCallInvoke(&fcinfo);
+
+	slot = MakeSingleTupleTableSlot(rsinfo.setDesc);
+
+	while (tuplestore_gettupleslot(rsinfo.setResult, true, false, slot))
+	{
+		HeapTuple	tuple = ExecFetchSlotTuple(slot);
+		Datum		values[DROPPED_OBJECTS_NATTS];
+		bool		nulls[DROPPED_OBJECTS_NATTS];
+		Oid			class_id;
+		char	   *objtype;
+
+		heap_deform_tuple(tuple, rsinfo.setDesc, values, nulls);
+
+		class_id = DatumGetObjectId(values[0]);
+
+		switch (class_id)
+		{
+			case ConstraintRelationId:
+				objtype = TextDatumGetCString(values[6]);
+				if (objtype != NULL && strcmp(objtype, "table constraint") == 0)
+				{
+					EventTriggerDropTableConstraint *obj = palloc(sizeof(EventTriggerDropTableConstraint));
```

Comment:
Can we do create functions for these objects? 

```
makeEventTriggerTableConstraint(schema, table, constraint_name, ...)
makeEventTriggerDropConstraint(...)
...
```

Gets a bit messy to have all the construction code in the switch statement.